매장수정 및 삭제 && 주변 매장 조회

TaehoLEEKR opened the pull request:
## 🎯 What is this PR
> 전체 적인 매장 API 구현 

## 📄 Changes Made
- 매장 업데이트 시 등록된 매장에 업데이트 진행
- 매장 조회 기능 ( GetAll, GetOne)
- 매장 삭제 기능 
- 매장 주변 위도,경도,반경nKm를 받아 주변 매장 조회 

## 📸 Screenshot
> 스크린 샷 첨부
![image](https://github.com/Siren-repo/Siren/assets/80689135/735934a6-27df-40c9-b212-17034e5c59ed)

## 🙋🏻‍ Review Point
- 매장의 등록할 때 권한을 로그인 했을 때의 권한으로 체크 해야 하여 추후 변경 해야 할 듯 합니다.
- 주변 매장의 조회는 프론트가 없어 직접 위도경도를 받아 체크하는 걸로 진행하였습니다.
## ✅ Test Checklist
- [x] 완료한 테스트 내용 작성

## 🔗 Reference
Issue {#6}

Commit:
chore: 기본 디펜던시 추가 및 프로파일 그룹별 설정

TaehoLEEKR committed:
feat : 매장 등록 기능 구현 (#001);

body :
Store Entity 구현 Store 등록 기본 정보 엔티티 생성
GlobalException 구현 domain 폴더 구조에서 외부로 꺼내 글로벌로 관리
GeoCoding 기능 구현 Geocoding API 호출 을 위한 apiClass 는 utils에서 관리
StoreController 와 StoreService 구현 매장을 등록하는 POST(/register) 구현

현재 로그인 기능이 없어 클라이언트에서 role을 직접 입력 받아 데이터를 전달

TaehoLEEKR committed:
Merge branch 'develop' into feature/001-매장등록

TaehoLEEKR committed:
style : PR 리뷰 피드백 변경(#6);

body :
Errorcode 폴더 위치 변경 > global
Entity Setter , AllArgsConstructor , 사용 지양
NoArgsConstructor access 제한
DTO validation 설정
Service @Transactional 추가

TaehoLEEKR committed:
feat : 매장 정보 수정 기능 구현(#6);

body :
ErrorCode 에러 추가
Store 매장 업데이트 기능 추가
storeId 를 pathValueable로 받아 업데이트를 진행

Store entity 에 update 메서드를 생성 하였고
objects.Objects.requireNonNullElse을 활용하여 null 일경우 기본 데이터를 가지고 없을 경우 업데이트를 진행

TEST : Mock 테스트 진행
정상 업데이트
비정상 업데이트
부분 업데이트 진행

TaehoLEEKR committed:
feat : 매장 정보 삭제 기능 추가 구현 및 테스트 진행(#6);

body :
ApiResponse 객체 추가
Store 매장 업데이트 기능 테스트 추가
storeId 를 pathValueable로 받아 업데이트 / 삭제 진행

Store entity 에 update 메서드를 생성 하였고
objects.Objects.requireNonNullElse을 활용하여 null 일경우 기본 데이터를 가지고 없을 경우 업데이트를 진행

TEST : Mock 테스트 진행
정상 업데이트
비정상 업데이트
부분 업데이트 진행

TaehoLEEKR committed:
feat : 매장 주변 매장조회 및 테스트(#6);

body :
불필요한 코드 삭제
주변 매장 조회 기능 추가

TEST : Mock 테스트 진행
정상 업데이트
비정상 업데이트
부분 업데이트 진행

yooyouny commented on src/main/java/com/devlop/siren/global/exception/ExceptionControllerAdvice.java:
```diff
@@ -0,0 +1,68 @@
+import javax.validation.ConstraintViolationException;
+
+@RestControllerAdvice
+public class ExceptionControllerAdvice extends ResponseEntityExceptionHandler {
```
이전 코드랑 바뀌어서 변경 된 내용으로 수정해야할 것 같습니다

yooyouny commented on src/main/java/com/devlop/siren/store/request/StoreUpdateRequest.java:
```diff
@@ -0,0 +1,27 @@
+import java.time.LocalDateTime;
+
+@Getter
+public class StoreUpdateRequest {
```
수정 request 객체는 따로 valid 검증은 안하시는 걸까요?

yooyouny commented on src/main/java/com/devlop/siren/store/service/StoreService.java:
```diff
@@ -0,0 +1,111 @@
+                return true;
+            } else {
+                log.error("저장 실패");
+                return false;
```
저장 실패로 false가 리턴되고 컨트롤러에서도 응답으로 false 값이 전달되는데 클라이언트가 false값을 받는게 의미가 있을까요? 그리고 false인데 상태코드는 200으로 전달되는것 같아서 실패면 응답이 ApiResponse.error() 로 전달되어야 하는건 아닌지 싶습니다.
밑에 지오코딩 파싱 에러도 GEOCODING_PARSING_ERROR를 만드시고 활용을 안하시는 것 같은데 예외를 발생시켜서 종료되도록 하는게 좋을방법이지 않을까 싶습니다.

yooyouny commented on src/main/java/com/devlop/siren/store/utils/GeocodingApi.java:
```diff
@@ -0,0 +1,23 @@
+
+import java.io.IOException;
+
+@Service
```
빈으로 등록하는거에 차이는 없지만 비즈니스 로직을 처리하는 빈인 @Service보다는 특정 기능을 수행하는 빈인 @Component가 더 자연스럽지 않을까 싶습니다

yooyouny reviewed (commented):
리뷰 남겨드렸습니다 수정이 필요한 부분은 예외처리 부분인것 같고 다른 부분에 대해서는 참고정도로 생각하셔도 될 것 같습니다 😀

Ho-Tea commented:
유연님이 리뷰하신것 제외하고는 따로 특이점을 발견하지는 못했습니다 Dto를 검증하는 부분에서 @Size대신 하이버네이트 어노테이션 @Length를 사용하셨는데 이부분은 크게 상관없을것 같습니다 고생하셨습니다 :)

TaehoLEEKR committed:
fix : 매장 관리 API(#6);

body :
불필요한 코드 삭제
TEST 코드 수정
에러 코드 수정

TaehoLEEKR committed:
Merge branch 'develop' of https://github.com/Siren-repo/Siren into feature/6-매장수정

# Conflicts:
#	build.gradle
#	src/main/java/com/devlop/siren/global/common/response/ApiResponse.java
#	src/main/java/com/devlop/siren/global/common/response/ResponseCode.java
#	src/main/java/com/devlop/siren/global/exception/ExceptionControllerAdvice.java
#	src/main/java/com/devlop/siren/global/exception/GlobalException.java
#	src/main/resources/application.yml
#	src/test/java/com/devlop/siren/SirenApplicationTests.java

TaehoLEEKR committed:
refactor : 매장 관리 API(#6);

body :
User의 권한에 따른 매장API 수정
매장 API 전체적인 TEST 코드 수정
조회 관련 TEST / CREAT,UPDATE,DELETE TEST 분리
